sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnInvokers.java

```diff
@@ -562,11 +565,11 @@ protected StackManipulation beforeDelegation(MethodDescription instrumentedMetho
         parameters.add(
             new StackManipulation.Compound(
                 pushExtraContextFactory,
-                MethodInvocation.invoke(EXTRA_CONTEXT_FACTORY_METHODS.get(param)),
+                MethodInvocation.invoke(getExtraContextFactoryMethod(param)),
                 // ExtraContextFactory.restrictionTracker() returns a RestrictionTracker,
                 // but the @ProcessElement method expects a concrete subtype of it.
                 // Insert a downcast.
-                (param == DoFnSignature.Parameter.RESTRICTION_TRACKER)
+                (param.equals(DoFnSignature.Parameter.restrictionTracker()))
```

Comment:
nit: generally prefer the more known/constant thing on the left to emphasize that there are no NPEs, etc.:
`DoFnSignature.Parameter.restrictionTracker().equals(param)`